postreise/plot/plot_scatter_capacity_vs_capacity_factor.py

```diff
@@ -0,0 +1,96 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from postreise.analyze.generation.capacity import (
+    get_capacity_by_resources,
+    get_capacity_factor_time_series,
+)
+from postreise.analyze.helpers import get_plant_id_for_resources_in_area
+from postreise.analyze.time import change_time_zone, slice_time_series
+
+
+def plot_scatter_capacity_vs_capacity_factor(
+    scenario,
+    area,
+    resources,
+    time_zone="utc",
+    time_range=None,
+    area_type=None,
+    between_time=None,
+    dayofweek=None,
+    markersize=50,
+    fontsize=20,
+    title=None,
+    show_plot=True,
+):
+    """Generate scatter plot with x-axis being capacity and y-axis being capacity
+    factor for a scenario in the specific area during any interested time range.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources.
+    :param str time_zone: new time zone, default to be *'utc'*.
+    :param tuple time_range: [start_timestamp, end_timestamp] where each time stamp
+        is pandas.Timestamp/numpy.datetime64/datetime.datetime. If None, the entire
+        time range is used for the given scenario.
+    :param str area_type: one of *'loadzone'*, *'state'*, *'state_abbr'*,
+        *'interconnect'*
+    :param list between_time: specify the start hour and end hour of each day
+        inclusively, default to None, which includes every hour of a day. Note that if
+        the end hour is set before the start hour, the complementary hours of a day are
+        picked.
+    :param set dayofweek: specify the interest days of week, which is a subset of
+        integers in [0, 7], default to None, which includes every day of a week.
+    :param float markersize: marker size, default to 50.
+    :param float fontsize: font size, default to 20.
+    :param str title: user specified figure title, default to None.
+    :param bool show_plot: show the plot or not, default to True.
+    :return: (*tuple*) -- the first entry is matplotlib.axes.Axes object of the plot,
+        the second entry is the capacity weighted average of capacity factors in the
+        plot.
```

Comment:
what do you mean by in the plot? Is that over the selected time range?

Reply:
Done.